lib/bitcrowd_ecto/repo.ex

```diff
@@ -133,9 +133,17 @@ defmodule BitcrowdEcto.Repo do
   end
 
   @doc false
-  @spec fetch(module, module, binary, keyword) :: fetch_result
-  def fetch(repo, module, id, opts) when is_atom(module) and is_binary(id) do
-    repo.fetch_by(module, [id: id], opts)
+  @spec fetch(module, module, any, keyword) :: fetch_result
+  def fetch(repo, module, id, opts) when is_atom(module) do
+     case module.__schema__(:primary_key) do
+      [pk] ->
+         repo.fetch_by(module, [{pk, id}], opts)
+
+      pks ->
+        raise ArgumentError,
+              "BitcrowdEcto.Repo.fetch/4 requires the schema #{inspect(module)} " <>
+                "to have exactly one primary key, got: #{inspect(pks)}"
+    end
```

Comment:
technically we could even allow multiple columns by checking that `id` is a list and zipping it with pks... but IMO we could leave this as-is for today and extend it when we need it